integrations/cavern-adapter/src/moneymarket.rs

```diff
@@ -0,0 +1,376 @@
+use abstract_money_market_standard::Identify;
+use abstract_sdk::{
+    core::objects::{ans_host::AnsHostError, AssetEntry, ContractEntry},
+    feature_objects::AnsHost,
+};
+use cosmwasm_std::{to_json_binary, QuerierWrapper, StdError};
+
+use crate::{AVAILABLE_CHAINS, CAVERN};
+
+// Source https://docs.rs/kujira/0.8.2/kujira/
+#[derive(Default)]
+pub struct Cavern {
+    pub oracle_contract: Option<Addr>,
+}
+
+impl Identify for Cavern {
+    fn name(&self) -> &'static str {
+        CAVERN
+    }
+    fn is_available_on(&self, chain_name: &str) -> bool {
+        AVAILABLE_CHAINS.contains(&chain_name)
+    }
+}
+
+#[cfg(feature = "full_integration")]
+use {
+    abstract_money_market_standard::{MoneyMarketCommand, MoneyMarketError},
+    cosmwasm_std::{wasm_execute, Addr, CosmosMsg, Decimal, Deps, Uint128},
+    cw_asset::{Asset, AssetInfo},
+};
+
+#[cfg(feature = "full_integration")]
+impl MoneyMarketCommand for Cavern {
+    fn fetch_data(
+        &mut self,
+        querier: &QuerierWrapper,
+        ans_host: &AnsHost,
+    ) -> Result<(), MoneyMarketError> {
+        let contract_entry = ContractEntry {
+            protocol: self.name().to_string(),
+            contract: "oracle".to_string(),
```

Comment:
We should have a document (readme?) containing all the required ANS entries for the protocol!

Reply:
Yes !